mm2src/coins/sia/spend_policy.rs

```diff
@@ -129,35 +127,73 @@ pub struct PolicyTypeThreshold {
 #[derive(Debug, Clone)]
 pub struct PolicyTypeUnlockConditions(UnlockCondition);
 
+// Sia v1 has theoretical support other key types via softforks
+// We only support ed25519 for now. No other type was ever implemented in Sia Go.
+#[derive(Debug, Clone)]
+pub struct UnlockKey {
+    algorithm: Specifier,
+    public_key: PublicKey,
+}
+
+impl Encodable for PublicKey {
+    fn encode(&self, encoder: &mut Encoder) { encoder.write_len_prefixed_bytes(&self.to_bytes()); }
```

Comment:
It feels like pubkeys shouldn't be len prefixed, since the length isn't variable.

Reply:
Has been refactor, but it is variable for some niche cases in their consensus mechanisms, but we only support typical ed25519. This logic is ported from:

https://github.com/SiaFoundation/core/blob/master/types/encoding.go